clients/src/main/java/org/apache/kafka/common/requests/AlterReplicaLogDirsResponse.java

```diff
@@ -38,10 +38,6 @@
 
     private final AlterReplicaLogDirsResponseData data;
 
-    public AlterReplicaLogDirsResponse(Struct struct) {
```

Comment:
Having this override seemed a bit error prone, and was causing failures for the `NetworkClientTest`. I opted to remove it entirely in favor of forcing the caller to specify the response version.